### PR TITLE
feat(troubleshoot): support for distribution kind

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What this PR does / why we need it:

Upgrades the troubleshoot package with kind support for distribution analyzer to better support all CMX distros.

#### Which issue(s) this PR fixes:

Fixes #
[sc-102282]

#### Does this PR introduce a user-facing change?

```release-note
The troubleshoot distribution analyzer now has support for distribution Kind.
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -419,3 +428,9 @@ replace (
 	golang.org/x/crypto v0.0.0-20220112180741-5e0467b6c7ce => github.com/dmacvicar/golang-x-crypto v0.0.0-20220126233154-a96af8f07497
 	gopkg.in/square/go-jose.v2 => gopkg.in/square/go-jose.v2 v2.2.2
 )
+
+// This is the only way I could get these two to play nice together
+replace (
+	github.com/replicatedhq/troubleshoot => github.com/replicatedhq/troubleshoot v0.86.0
+	github.com/vmware-tanzu/velero => github.com/vmware-tanzu/velero v1.10.0-rc.1.0.20240408090904-54462c4f7bb5
+)
```
Comment (emosbaugh) [inline go.mod:436]: this was weird

File: pkg/docker/registry/auth.go [hunk 2]
```diff
@@ -11,8 +11,8 @@ import (
 
 	"github.com/containers/image/v5/pkg/docker/config"
 	"github.com/containers/image/v5/types"
-	"github.com/distribution/distribution/v3/registry/client/auth/challenge"
```
Comment (emosbaugh) [inline pkg/docker/registry/auth.go:14]: this package is now internal

Comment (cbodonnell) [reply pkg/docker/registry/auth.go:14]: what's the reason behind needing to make this one internal?

Comment (emosbaugh) [reply pkg/docker/registry/auth.go:14]: github.com/distribution/distribution has made it internal so we can no longer import it

https://github.com/distribution/distribution/tree/bc6e81e1b9a8017a66fc56a55913b92930941feb/internal/client/auth/challenge

**Review by cbodonnell — APPROVED** [review]

**Pull request merged**